pkg/provisioner/ironic/ironic.go

```diff
@@ -1013,6 +1024,16 @@ func (p *ironicProvisioner) Provision(getUserData provisioner.UserDataSource) (r
 	case nodes.Manageable:
 		return p.startProvisioning(ironicNode, checksum, getUserData)
 
+	case nodes.CleanFail:
+		if ironicNode.LastError == "" {
+			p.log.Info("failed but error message not available")
+			result.Dirty = true
+			return result, nil
+		}
+		p.log.Info("found error", "msg", ironicNode.LastError)
+		result.ErrorMessage = fmt.Sprintf("Hardware configuration failed. Error: %s", ironicNode.LastError)
+		return result, nil
```

Comment:
This is consistent with how we handle some other errors, but it's going to put the host into a state the user can't recover from easily. I don't have a good answer for that, but thought I should point it out. I think if we agree on the rest of this change we can deal with the error handling in a separate PR.

Reply:
Just recording my agreement with Doug (I'm fine with a separate PR).